internal/static/static.go

```diff
@@ -0,0 +1,75 @@
+package static
+
+import (
+	"go/types"
+
+	"github.com/dogmatiq/configkit"
+	"golang.org/x/tools/go/packages"
+	"golang.org/x/tools/go/ssa"
+	"golang.org/x/tools/go/ssa/ssautil"
+)
+
+const (
+	// dogmaPkgPath is the full path of dogma package.
+	dogmaPkgPath = "github.com/dogmatiq/dogma"
+)
+
+// FromPackages loads the list of configkit.Application from the given list
+// of packages.Package items.
+func FromPackages(pkgs []*packages.Package) []configkit.Application {
+	prog, packages := ssautil.AllPackages(pkgs, ssa.SanityCheckFunctions)
+	prog.Build()
+
+	if dogmaPkg := prog.ImportedPackage(dogmaPkgPath); dogmaPkg != nil {
+		return fromSSAPackages(dogmaPkg, packages)
+	}
+
+	// If Dogma package is not found as an import, none of the packages in the
+	// built application implement dogma.Application interface that requires the
+	// import of the github.com/dogmatiq/dogma package.
+	return nil
+}
+
+// fromSSAPackages loads the list of configkit.Application from the given list
+// of ssa.Package items.
+func fromSSAPackages(dogmaPkg *ssa.Package, pkgs []*ssa.Package) []configkit.Application {
+	var result []configkit.Application
+
+	for _, pkg := range pkgs {
+		// The package can be returned as nil by ssautil.AllPackages() if it
+		// previously contained errors in types.Package.Errors.
+		if pkg == nil {
+			continue
+		}
+
+		a := dogmaPkg.Pkg.Scope().Lookup("Application")
+		iface := a.Type().Underlying().(*types.Interface)
```

Comment:
This has been moved to `FromPackages()` function in https://github.com/dogmatiq/configkit/pull/118/commits/ae646d88b13b26bb47f25097dcab1ca084511547. The entire `fromSSAPackages()` function was merged into `FromPackages()` as there was not much left of the former. In addition , if left unmerged it would have some uncovered sections of code.